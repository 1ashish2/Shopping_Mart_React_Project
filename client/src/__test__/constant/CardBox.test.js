import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import CardBox from "../../constant/CardBox.constant";
import { MemoryRouter } from "react-router-dom";

afterEach(() => {
  cleanup();
});

describe("Button Component", () => {
  const handleAddProduct = jest.fn();
  // const handleClick=jest.fn()
  const detail = {
    name: "Fresho Kiwi - Green, 3 pcs",
    imageURL: "/static/images/products/fruit-n-veg/kiwi-green.jpg",
    description:
      "Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.",
    price: 87,
    stock: 90,
    totalStock: 90,
    amountQty: 3,
    qty: "pcs",
    category: "5b6899953d1a866534f516e2",
    sku: "fnw-kiwi-3",
    id: "5b6c6a7f01a7c38429530883",
  };

  it("matches snapshot", () => {
    const { asFragment } = render(
      <CardBox detail={detail} handleAddProducte={handleAddProduct} />,
      { wrapper: MemoryRouter }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Button Text", () => {
    render(<CardBox detail={detail} handleAddProduct={handleAddProduct} />, {
      wrapper: MemoryRouter,
    });
    const button = screen.getByRole("button");
    // expect(button).toHaveTextContent("Add to Cart");
    fireEvent.click(button);
  });

  test("should be able to type name input field", () => {
    render(<CardBox detail={detail} handleAddProduct={handleAddProduct} />, {
      wrapper: MemoryRouter,
    });
    const inputEl = screen.getByTestId("input");
    fireEvent.change(inputEl, { target: { value: 12 } });
    expect(inputEl.value).toBe("12");
  });
});
