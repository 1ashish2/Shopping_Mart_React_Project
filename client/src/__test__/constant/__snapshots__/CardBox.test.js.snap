// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="cart-main card"
  >
    <a
      href="/productDetail"
    >
      <img
        alt="Fresho Kiwi - Green, 3 pcs"
        class="card-img-top"
        src="/static/images/products/fruit-n-veg/kiwi-green.jpg"
      />
    </a>
    <div
      class="card-body"
    >
      <a
        class="text-dark link-cart-title"
        href="/productDetail"
      >
        <div
          class="cart-title card-title h5"
        >
          <strong>
            Fresho Kiwi - Green, 3 pcs
          </strong>
        </div>
      </a>
      <div
        class="cardBoxPrice-container"
      >
        <span
          class="cardBox-price"
        >
          Price â‚¹87
        </span>
        <span
          class="cardBox-price"
        >
          Per 3 pcs
        </span>
        <div
          class="card-box-input-button"
        >
          <input
            aria-label="number of quantity"
            class="card-box-input-field"
            data-testid="input"
            max="90"
            min="0"
            type="number"
            value="0"
          />
          <div>
            <button
              class="btn"
              data-testid="button"
            >
              Add to Cart
            </button>
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
